### PR TITLE
Remove metrics spi dependency

### LINKED ISSUE #12773: Quarkus Kubernetes extension conflict with Micrometer extension
Quarkus 1.9.1.CR1

**Describe the bug**
When I add Micrometer extension to existing Quarkus project with S3 extension, I am getting below compilation error
`java
Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors [ERROR] [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: Could not resolve artifact org.eclipse.microprofile.metrics:microprofile-metrics-api::jar among the runtime dependencies of the application [ERROR] at 
`
**Expected behavior**
I should be able to compile the project successfully with both S3 extension and MicroMeter extension

**Actual behavior**
Compilation Error

**To Reproduce**

1. Create a Quarkus Projecct with S3 extension
2. Compile build and test (Success)
3. Add Micrometer extension to the project
4. Compile Project (Failure)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: mac os & Linux
 - Output of `java -version`:  Java 11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.9.1.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):

### PR DESCRIPTION
**Pull request by ebullient**
Resolves #12773

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/pom.xml [hunk 1]
```diff
@@ -29,10 +29,6 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-kubernetes-client-deployment-internal</artifactId>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-smallrye-metrics-spi</artifactId>
-        </dependency>
```
Comment (geoand) [inline extensions/kubernetes/vanilla/deployment/pom.xml:35]: Can you explain this change?

1) I don't get it why it's needed (is this simply left over by mistake?)
2) If this is not left over by mistake, I am pretty sure things will break

Comment (ebullient) [reply extensions/kubernetes/vanilla/deployment/pom.xml:35]: I think that when I went through and made metrics changes in 1.7 (to remove direct usage of the old metrics SPI in favor of the new Metrics capability checks in core), I forgot to remove this.

Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/pom.xml:35]: OK, let's see if CI passes

Comment (ebullient) [reply extensions/kubernetes/vanilla/deployment/pom.xml:35]: Switch to new API: 
https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java#L103
to retrieve metrics path: 
https://github.com/quarkusio/quarkus/blob/cbe8d21b9b97efdb047e8a1abe8a55104688029a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java#L296

That was all the SPI was used for..

**Review by geoand — APPROVED** [review]
Approving pending CI validation :)

**Comment by gsmet** [comment]
Adding it as a good backport candidate for 1.9.1.Final.

**Pull request merged**